src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -289,12 +287,72 @@ def __init__(
             "stream_cache_path": stream_cache_path,
         }
 
+    def read_nwb_file(self, file_path: str, stream_mode: Literal[None, "ffspec", "ros3"] = None, stream_cache_path: Optional[str] = None) -> NWBFile:
+        """
+        Read an NWB file and return the NWBFile object.
+
+        Parameters
+        ----------
+        file_path : str
+            The path to the NWB file.
+        stream_mode : str, optional
+            The streaming mode to use. Default is None.
```

Comment:
This, as well as the matching part of the class definition, aren't true anymore if we start using `Literal`

```suggestion
        stream_mode : "fsspec" or "ros3", optional
            The streaming mode to use. Default assumes the file is on the local disk.
```